doc/developer/code-coverage.md

```diff
@@ -0,0 +1,50 @@
+# Code Coverage for Pull Requests
+
+## Motivation
+
+A full [code coverage report](http://65.109.125.29/coverage_01879007-43c4-49db-a81d-a44e93dfb55a/) allows you to check for each part of the code if there is a test that currently covers it by executing the relevant lines of code. Unfortunately the motivation to increase code coverage using full reports is low, and the effort to understand why no existing test covers the code in question is high.
+
+Thus to prevent regressions and new bugs we instead want to ensure that new and changed productive code in a pull request (PR) is adequately tested. This makes sense both when code is refactored as well as when new functionality is added. For each PR you or someone from the QA team can use the [Coverage](https://buildkite.com/materialize/coverage/builds) Buildkite pipeline to trigger coverage for it.
```

Comment:
~~If one triggers just the coverage evaluation, is this an expensive step? If not, is there a reason why we don't automatically execute this step?~~

Ok, answered below 😀